src/components/UntrustedBalance.tsx

```diff
@@ -105,31 +142,57 @@ export const UntrustedBalance = ({
 
   return (
     <>
-      {untrustedAssets.data.map((asset: Asset) => (
-        <BalanceRow
-          activeAsset={activeAsset.asset}
-          key={asset.assetString}
-          asset={asset}
-          onChange={(e) =>
-            handleActionChange({ actionId: e.target.value, asset })
-          }
-        >
-          <TextButton
-            onClick={() =>
-              handleActionChange({
-                actionId: AssetActionId.TRUST_ASSET,
-                asset,
-              })
-            }
-            disabled={
-              Boolean(activeAsset.asset) ||
-              trustAsset.status === ActionStatus.PENDING
+      {untrustedAssets.data.map((asset: Asset) =>
+        asset.isUntrusted ? (
+          // Untrusted
+          <BalanceRow
+            activeAsset={activeAsset.asset}
+            key={asset.assetString}
+            asset={asset}
+            onChange={(e) =>
+              handleActionChange({ actionId: e.target.value, asset })
             }
           >
-            Trust asset
-          </TextButton>
-        </BalanceRow>
-      ))}
+            <TextButton
+              onClick={() =>
+                handleActionChange({
+                  actionId: AssetActionId.TRUST_ASSET,
+                  asset,
+                })
+              }
+              disabled={
+                account.isUnfunded ||
+                Boolean(activeAsset.asset) ||
+                trustAsset.status === ActionStatus.PENDING
+              }
```

Comment:
v small - i liked what you did with `formIsEmpty` above. Since you have 2 similar disabled statements, might be nice to do:
```
const assetNotFound = Boolean(activeAsset.asset) || trustAsset.status === ActionStatus.PENDING; 
// with a more descriptive var name lol

disabled = {account.isUnfunded || assetNotFound}
```

Reply:
Yeah, this one got a little long. 😅 A variable there will make it much more clear. 👍